src/org/jetbrains/plugins/scala/caches/HydraArtifactsCache.scala

```diff
@@ -0,0 +1,55 @@
+package org.jetbrains.plugins.scala.caches
+
+import java.io.File
+import java.util.concurrent.ConcurrentHashMap
+import org.jetbrains.plugins.scala.project.template.FileExt
+import com.intellij.openapi.diagnostic.Logger
+import org.jetbrains.plugins.scala.project.template.Downloader
+
+/**
+  * @author Maris Alexandru
+  */
+object HydraArtifactsCache {
+  private val Log = Logger.getInstance(this.getClass)
+  private val cache = new ConcurrentHashMap[(String, String), Seq[String]]()
+  private val SplitRegex = "\\* Attributed\\(|\\)".r
+  private val ArtifactsRegex = "\\* Attributed\\(.*\\)".r
+  private val GroupId = "com.triplequote"
+
+  def getOrDownload(scalaVersion: String, hydraVersion: String, listener: (String) => Unit): Seq[String] = {
+    val artifacts = cache.getOrDefault((scalaVersion, hydraVersion), Seq.empty)
+
+    if (artifacts.isEmpty) {
+      val sbtBufferedOutput = new StringBuilder
+      Downloader.downloadHydra(s"${scalaVersion}_$hydraVersion", (text: String) => {
+        sbtBufferedOutput.append(text)
+        listener(text)
+      })
+      Log.info(sbtBufferedOutput.toString())
```

Comment:
Add a message before appending the log (this is useful in many ways. For instance, what if the log is empty? ;))